dvol_python/test_dvol.py

```diff
@@ -55,43 +80,52 @@ class VoluminousTests(TestCase):
     def setUp(self):
         self.tmpdir = FilePath(self.mktemp())
         self.tmpdir.makedirs()
-        self.patch(Voluminous, "lockFactory", NullLock)
 
     def test_create_volume(self):
-        # TODO test volume names with '/' in them - they should not end up
-        # making nested heirarchy
         dvol = VoluminousOptions()
-        dvol.parseOptions(["-p", self.tmpdir.path, "init", "foo"])
+        dvol.parseOptions(ARGS + ["-p", self.tmpdir.path, "init", "foo"])
         self.assertTrue(self.tmpdir.child("foo").exists())
         self.assertTrue(self.tmpdir.child("foo").child("branches")
                 .child("master").exists())
-        self.assertEqual(dvol.voluminous.getOutput(),
-                ["Created volume foo", "Created branch foo/master"])
+        self.assertEqual(dvol.voluminous.getOutput()[-1],
+                "Created volume foo\nCreated branch foo/master")
 
     def test_create_volume_already_exists(self):
         dvol = VoluminousOptions()
-        dvol.parseOptions(["-p", self.tmpdir.path, "init", "foo"])
-        self.assertRaises(VolumeAlreadyExists,
-                dvol.parseOptions, ["-p", self.tmpdir.path, "init", "foo"])
-        self.assertEqual(dvol.voluminous.getOutput(),
-                ["Error: volume foo already exists"])
+        # Create the repository twice, second time should have the error
+        dvol.parseOptions(ARGS + ["-p", self.tmpdir.path, "init", "foo"])
+        dvol.parseOptions(ARGS + ["-p", self.tmpdir.path, "init", "foo"])
+        self.assertEqual(dvol.voluminous.getOutput()[-1],
+                "Error: volume foo already exists")
+
+    def test_create_volume_with_path_seperator(self):
+        dvol = VoluminousOptions()
+        dvol.parseOptions(ARGS + ["-p", self.tmpdir.path, "init", "foo/bar"])
+        self.assertEqual(dvol.voluminous.getOutput()[-1],
+                "Error: foo/bar is not a valid name")
```

Comment:
I'd rather we didn't. Here's why:

* There's a general class of bugs here (e.g. `.` and `..` are equally invalid names that will raise tracebacks that are fixed by this code but are untested). I don't really see why we should fix them one at a time like pets when we could eliminate them en masse
* We're looking at a Go implementation. That means that the tests are the main value of the Python code base.
* We already have tests that use Hypothesis in this module. That whole raft of work is mostly adapting from those, and will take only a little longer than we've spent debating this issue.
* We know the definition of "valid" and "invalid" is going to change, so we should abstract away those concepts, rather than hard-coding them in place
* If we can get a computer to do our work for us, we should.

However, preventing a traceback in this one case _is_ an incremental improvement. In the spirit of moving fast and breaking things, you can land without Hypothesis, and without covering the other traceback-raising cases.

I'm going to make some other comments that will need to be addressed before landing though.